CDTDatastore/CDTReplicator/CDTReplicator.m

```diff
@@ -109,26 +110,51 @@ + (NSString *)stringForReplicatorState:(CDTReplicatorState)state
     }
 }
 
++ (CDTReplicator *)replicatorWithConfiguration:(CDTAbstractReplication *)configuration
```

Comment:
which is the problem. The class is the replicator and without this exposed we call `init` instead. See https://github.com/cloudant/CDTDatastore/pull/309/files#diff-4060e2491f9dfa52069155a3851436f3R84 so we should either be using this method or it should be removed.